courses/utils_test.py

```diff
@@ -209,3 +258,50 @@ def test_sync_course_runs(settings, mocker, mocked_api_response, expect_success)
     else:
         assert success_count == 0
         assert failure_count == 1
+
+
+def test_ensure_course_run_grade():
+    """Test that ensure_course_run_grade works as expected"""
+    run = CourseRunFactory.create()
+    user = UserFactory()
+    edx_grade = MagicMock(percent=0.5, passed=True, letter_grade="A")
+    grade, created, updated = ensure_course_run_grade(user, run, edx_grade)
+    assert grade.course_run == run
+    assert grade.user == user
+    assert grade.grade == 0.5
+    assert grade.letter_grade == "A"
+    assert grade.passed is True
+    assert grade.set_by_admin is False
+    assert created is True
+    assert updated is False
+
+
+def test_ensure_course_run_grade_update():
+    """Test that ensure_course_run_grade works as expected when an update is indicated"""
+    run = CourseRunFactory.create()
+    user = UserFactory()
+    edx_grade = MagicMock(percent=0.5, passed=True, letter_grade="A")
+    grade, created, updated = ensure_course_run_grade(user, run, edx_grade)
+
+    assert grade.course_run == run
+    assert grade.user == user
+    assert grade.grade == 0.5
+    assert grade.letter_grade == "A"
+    assert grade.passed is True
+    assert grade.set_by_admin is False
+    assert created is True
+    assert updated is False
```

Comment:
Same comment as above. Please just create the necessary object(s) directly instead of running ensure_course_run_grade to create them, and remove any testing assertions that are repeated from the test above (everything from line 286 to 293)